operator-common/src/main/java/io/strimzi/operator/common/model/Labels.java

```diff
@@ -361,4 +393,24 @@ public int hashCode() {
     public String toString() {
         return "Labels" + labels;
     }
+
+    /**
+     * @param resource        a resource with metadata
+     * @param applicationName the name of the application of the component
+     * @param managedBy       a resource with meta
+     * @return The default Labels set
+     */
+    public static Labels generateDefaultLabels(HasMetadata resource, String applicationName, String managedBy) {
+        String instanceName = resource.getMetadata().getName();
```

Comment:
I think we need to sanitize this for length

EDIT: I can see you do that in the `with...()` methods. That means it is done multiple times and here were are vulnerable to someone adding new label without sanitizing it. Maybe you have a reson to keep it in the `with...()` methods. But it might be good to leave here a comment or something.

Reply:
Some comment along the lines of :
```
// This string may be truncated for sanitization purposes to be compatible with Kubernetes
```
, or am I misunderstanding your suggestion?

Reply:
I would maybe phrase it more as in instruction for the next developer:

```
// When adding new labels, make sure the names of any resources are truncated for sanitization purposes to be compatible with Kubernetes => Labels can be only short
```